src/sql/statements.cpp

```diff
@@ -2,19 +2,36 @@
 
 namespace hsql {
 
+  // KeyConstraints
+  TableKeyConstraint::TableKeyConstraint(ConstraintType type, std::vector<char*>* columnNames) :
+    type(type),
+    columnNames(columnNames) {};
+
+  TableKeyConstraint::~TableKeyConstraint() {
+    for (char* def : *columnNames) {
+      delete def;
+    }
+    delete columnNames;
+  }
   // ColumnDefinition
-  ColumnDefinition::ColumnDefinition(char* name, ColumnType type, bool nullable) :
+  ColumnDefinition::ColumnDefinition(char* name, ColumnType type, bool nullable, ConstraintType constraintType) :
     name(name),
     type(type),
-    nullable(nullable) {};
+    nullable(nullable),
+    constraintType(constraintType) {};
 
   ColumnDefinition::~ColumnDefinition() {
     free(name);
   }
 
-  ColumnType::ColumnType(DataType data_type, int64_t length) :
+  ColumnType::ColumnType(DataType data_type, int64_t length, DecimalSpecification decimal_specification) :
     data_type(data_type),
-    length(length) {};
+    length(length),
+    decimal_specification(decimal_specification) {};
+
+  DecimalSpecification::DecimalSpecification(int64_t precision, int64_t scale) :
```

Comment:
TimeSpecification is missing, isn't it?

Reply:
It is now introduced in https://github.com/hyrise/sql-parser/pull/177/commits/83d45ac410bdf109e981f1a4b8bb596a409aa13c